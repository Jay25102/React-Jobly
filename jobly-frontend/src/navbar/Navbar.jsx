import React, { useContext } from "react";
import { NavLink } from "react-router-dom";
import UserContext from "../forms/UserContext";

/**
 * Site-wide navbar. Shows different options when logged in/out.
 * Logs user out using the logout function defined in App
 */
function Navbar({ logout }) {
    const { currentUser } = useContext(UserContext);
    console.debug("Navigation", "currentUser=", currentUser);

    function loggedInNav() {
        return (
            <div>
                <ul className="navbar-nav ml-auto">
                    <li className="nav-item mr-4">
                        <NavLink className="nav-link" to="/companies">Companies</NavLink>
                    </li>
                    <li className="nav-item mr-4">
                        <NavLink className="nav-link" to="/jobs">Jobs</NavLink>
                    </li>
                    <li className="nav-item mr-4">
                        <NavLink className="nav-link" to="/profile">Profile</NavLink>
                    </li>
                    <li className="nav-item mr-4">
                        <NavLink className="nav-link" to="/" onClick={logout}>Log out ({currentUser.first_name || currentUser.username})</NavLink>
                    </li>
                </ul>
            </div>
        )
    }

    function loggedOutNav() {
        return (
            <div>
                <ul className="navbar-nav ml-auto">
                    <li className="nav-item mr-4">
                        <NavLink className="nav-link" to="/login">Login</NavLink>
                    </li>
                    <li className="nav-item mr-4">
                        <NavLink className="nav-link" to="/signup">Signup</NavLink>
                    </li>
                </ul>
            </div>
        )
    }

    return (
        <nav className="Navbar navbar navbar-expand-md">
            <NavLink className="navbar-brand" to="/">Homepage</NavLink>
            {currentUser ? loggedInNav() : loggedOutNav()}
        </nav>
    )
}

export default Navbar;