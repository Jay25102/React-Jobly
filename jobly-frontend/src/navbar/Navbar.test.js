import React from "react";
import { render } from "@testing-library/react";
import Navbar from "./Navbar";
import { MemoryRouter } from "react-router-dom";
import { UserProvider } from "../testUtils";

describe("Navbar", function() {
    test("it renders without crashing", function() {
        render(
            <MemoryRouter>
                <UserProvider>
                    <Navbar/>
                </UserProvider>
            </MemoryRouter>
        );
    });
    test("it matches snapshot logged in", function() {
        const { asFragment } = render(
            <MemoryRouter>
                <UserProvider>
                    <Navbar/>
                </UserProvider>
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });
    test("it matches snapshot logged out", function() {
        const { asFragment } = render(
            <MemoryRouter>
                <UserProvider currentUser={null}>
                    <Navbar/>
                </UserProvider>
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});