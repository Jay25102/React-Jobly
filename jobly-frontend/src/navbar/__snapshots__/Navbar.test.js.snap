// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar it matches snapshot logged in 1`] = `
<DocumentFragment>
  <div>
    <a
      aria-current="page"
      class="active"
      href="/"
    >
      Homepage
    </a>
    <div>
      <a
        class=""
        href="/companies"
      >
        Companies
      </a>
      <a
        class=""
        href="/jobs"
      >
        Jobs
      </a>
      <a
        class=""
        href="/profile"
      >
        Profile
      </a>
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Log out testfirst
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Navbar it matches snapshot logged out 1`] = `
<DocumentFragment>
  <div>
    <a
      aria-current="page"
      class="active"
      href="/"
    >
      Homepage
    </a>
    <div>
      <a
        class=""
        href="/login"
      >
        Login
      </a>
      <a
        class=""
        href="/signup"
      >
        Signup
      </a>
    </div>
  </div>
</DocumentFragment>
`;
